### PR TITLE
Iteration0 integration

### PR DESCRIPTION
**Pull request by DougWelchons**
Here are all the changes that we made for integration of our iteration 0 files. we will need to update our test files to reflect the changes, and time permitting we will need to do some refactoring as well.

### CODE REVIEW

File: lib/sales_engine.rb [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+require_relative 'item_repository'
+
 class SalesEngine
-attr_reader :items, :merchants
+  attr_reader :items, :merchants
 
   def initialize(arg1)
-    @items = CSV.readlines(arg1[:items], headers: true, header_converters: :symbol)
-    @merchants = CSV.readlines(arg1[:merchants], headers: true, header_converters: :symbol)
+  @items = things(CSV.readlines(arg1[:items], headers: true, header_converters: :symbol))
```
Comment (DougWelchons) [inline lib/sales_engine.rb]: corrected the indention

File: lib/merchant_repository.rb [hunk 2]
```diff
@@ -8,48 +9,67 @@ def initialize(merchants)
...
-    attributes[:id] = max_item_id + 1
-    attributes[:name] = name.to_s
-    attributes[:created_at] = Time.strftime("%Y-%m-%d")
-    attributes[:updated_at] = Time.now.to_s #replace with above?
+    # attributes[:id] = max_item_id + 1
+    # attributes[:name] = name.to_s
+    # attributes[:created_at] = Time.strftime("%Y-%m-%d")
+    # attributes[:updated_at] = Time.now.to_s #replace with above?
+
+
```
Comment (DougWelchons) [inline lib/merchant_repository.rb]: removed extra lines

File: lib/item_repository.rb [hunk 3]
```diff
@@ -1,69 +1,86 @@
...
-    attributes[:updated_at] = Time.now.to_s
+    @items.push(Item.new({
+                          id: (max_item_id + 1),
+                          name: attributes[:name],
+                          description: attributes[:description],
+                          unit_price: attributes[:unit_price],
+                          merchant_id: attributes[:merchant_id],
+                          created_at: attributes[:created_at] = Time.now.to_s,
+                          updated_at: attributes[:updated_at] = Time.now.to_s
+                        }))
```
Comment (DougWelchons) [inline lib/item_repository.rb]: corrected

File: lib/merchant_repository.rb [hunk 4]
```diff
@@ -8,48 +9,67 @@ def initialize(merchants)
...
     merchants.find do |merchant|
-      merchant[:id].to_i == id
+      merchant.id.to_i == id
     end
   end
 
   def find_by_name(name)
     merchants.find do |merchant|
-      merchant[:name] == name
+      merchant.name.downcase == name.downcase
```
Comment (Diana20920) [inline lib/merchant_repository.rb]: @DougWelchons `#casecmp` is the one I was trying to use but couldn't quite figure out 🤓 I will see if doing `merchant.name.casecmp(name)` will work, but I may reach out for help.

File: lib/item.rb [hunk 5]
```diff
@@ -15,13 +15,13 @@ def initialize(data)
     @id           = data[:id].to_i
     @name         = data[:name]
     @description  = data[:description]
-    @unit_price   = BigDecimal.new(data[:unit_price])
+    @unit_price   = BigDecimal.new((data[:unit_price].to_i)) / 100
```
[the remark this answers is not in the document]
Comment (Diana20920) [reply lib/item.rb]: Do we want to make the Lint/BigDecimalNew change suggested by Hound? If it even keeps the Spec Harness green, of course.

File: lib/sales_engine.rb [hunk 6]
```diff
@@ -1,13 +1,34 @@
...
+
 class SalesEngine
-attr_reader :items, :merchants
+  attr_reader :items, :merchants
 
   def initialize(arg1)
-    @items = CSV.readlines(arg1[:items], headers: true, header_converters: :symbol)
-    @merchants = CSV.readlines(arg1[:merchants], headers: true, header_converters: :symbol)
+    @items = things(CSV.readlines(arg1[:items], headers: true, header_converters: :symbol))
+    @merchants = people(CSV.readlines(arg1[:merchants], headers: true, header_converters: :symbol))
```
[the remark this answers is not in the document]
Comment (brisag) [reply lib/sales_engine.rb:12]: Cant really shorten it, unless we create a helper file that holds "headers: true, header_converters: :symbol))"

**Comment by Diana20920** [comment]
This is amazing. Spec Harness was so confusing at first but it was sure nice to see it all turn to green eventually.
There are two unresolved conversations that I wanted to get your opinion/thoughts on. Perhaps we can leave them for refactoring day since I have noted them as pending tasks on my to-do tracker. 

Otherwise, this is good to merge!

File: lib/merchant_repository.rb (hunk 4 above)
Comment (DougWelchons) [reply lib/merchant_repository.rb]: I just tried to implement the `#casecmp`, and it appears to have worked. I will push that up and add it to the PR before we merge.

**Comment by DougWelchons** [comment]
@Diana20920, I saw your note on the #casecmp and made that change. but i didn't see another comment from you, what was the second issue?

File: lib/item.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (DougWelchons) [reply lib/item.rb]: I have also made this change and it still passes the SpecHarness, so 👍 on this one.

**Comment by avjohnston** [comment]
Looks awesome! There is some slight refactoring we could do, but like you said lets gets those tests finished for now and come back to the refactoring if we have time.

**Pull request merged**